docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,433 @@
+
+const config: HardhatUserConfig = {
+    zksolc: {
+        version: '1.3.7',
+        compilerSource: 'binary',
+        settings: {
+            optimizer: {
+                enabled: true,
+            },
+        },
+    },
+    defaultNetwork: 'zkSyncNetwork',
+    networks: {
+        goerli: {
+            zksync: false,
+            url: 'http://localhost:3050',
+        },
+        zkSyncNetwork: {
+            zksync: true,
+            ethNetwork: 'goerli',
+            url: 'http://localhost:8545',
+        },
+    },
+    solidity: {
+        version: '0.8.19',
+    },
+};
+
+export default config;
+
+```
+
+# Deploying Proxies
+
+The plugin supports two types of proxies: Transparent upgradable proxies and beacon proxies.
+
+Upgradability methods are all part of the `zkUpgrades` property in the HardhatRuntimeEnvironment and you only need to interact with it in order to deploy or upgrade your contracts.
+
+Let's see how you can use proxies in more detail:
+
+For the following examples, we will use the simple "Box" smart contract:
+
+```typescript
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+
+contract Box is Initializable{
+    uint256 private value;
+    uint256 private secondValue;
+    uint256 private thirdValue;
+
+    function initialize(uint256 initValue) public initializer {
+        value = initValue;
+    }
+
+    // Reads the last stored value
+    function retrieve() public view returns (uint256) {
+        return value;
+    }
+
+    // Stores a new value in the contract
+    function store(uint256 newValue) public {
+        value = newValue;
+        emit ValueChanged(newValue);
+    }
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+}
+
+```
+
+
+## Transparent upgradable proxies
+
+Transparent upgradable proxies provide a way to upgrade a smart contract without changing its address or requiring any change in the contract's interaction code. With transparent proxies, a contract's address is owned by a proxy contract, which forwards all calls to the actual contract implementation. When a new implementation is deployed, the proxy can be upgraded to point to the new implementation, allowing for seamless upgrades without requiring changes to the contract's interaction code.
+
+To deploy a simple upgradable contract on the zkSync local setup, first create a test wallet and add it to the new Deployer.
+
+```typescript
+
+  const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+  const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+  const deployer = new Deployer(hre, zkWallet);
+```
+
+After that, simply load the Box artifact and call the `deployProxy` method from the `zkUpgrades` hre property.
+
+```typescript
+  const contract = await deployer.loadArtifact(contractName);
```

Comment:
are we assuming the user knows how to create the contract? i.e. `contractName` versus `box` here?

Reply:
Is it worth having a preliminary step that sets this up after ## Configuration? as it is used all the way through..

Reply:
Added an explanation right below the Box contract on what is expected before it is used in our script, along with linked example on how to do it.

Reply:
If you think it is necessary, I can also add explicit `yarn hardhat compile` command below it, but I would say linked page explains pretty well how to get started.

Reply:
SGTM